taf/updater/updater.py

```diff
@@ -1305,22 +1305,25 @@ def _validate_authentication_repository(
         targets = validation_auth_repo.get_json(commits[-1], "metadata/targets.json")
         test_repo = "test-auth-repo" in targets["signed"]["targets"]
         if test_repo and expected_repo_type != UpdateType.TEST:
-            error_msg = UpdateFailedError(
+            error_msg = (
                 f"Repository {users_auth_repo.name} is a test repository. "
                 'Call update with "--expected-repo-type" test to update a test '
                 "repository"
             )
         elif not test_repo and expected_repo_type == UpdateType.TEST:
-            error_msg = UpdateFailedError(
+            error_msg = (
                 f"Repository {users_auth_repo.name} is not a test repository,"
                 ' but update was called with the "--expected-repo-type" test'
             )
 
     # always cleanup repository updater
-    repository_updater.update_handler.cleanup()
+    try:
+        repository_updater.update_handler.cleanup()
```

Comment:
This is called even when no errors happen? So if cleanup fails, should update as a whole raise an error? Without this fix, update just fails, so this certainly is not a regression. Should this be a warning like when pip prints that we should upgrade?